Description: from ocaml-num/1.6, nums.cmxa is not shipped anymore and
 compilation has to be handled differently, see the page of the Debian bug.
Author: Pierre Gruet <pgt@debian.org>
Bug-Debian: https://bugs.debian.org/1111281
Forwarded: https://gitlab.com/scilab/scilab/-/merge_requests/1451
Last-Update: 2025-08-18

--- a/scilab/modules/scicos/Makefile.modelica.am
+++ b/scilab/modules/scicos/Makefile.modelica.am
@@ -111,19 +111,19 @@
 # Build the binaries, disabling parallel building as ocamldep ordering seems to fail on some systems
 
 modelicac: $(modelicac_SOURCES) $(MLS_modelicac)
-	$(OCAMLOPT) -o $@ -I $(BASE_PATH_MODELICAC) $(OCAMLOPTFLAGS) $(OCAMLLIBS:.cma=.cmxa) $(MLS_modelicac)
+	ocamlfind ocamlopt -package num -linkpkg -o $@ -I $(BASE_PATH_MODELICAC) $(OCAMLOPTFLAGS) $(MLS_modelicac)
 
 modelicac.ocamlc: $(modelicac_SOURCES) $(MLS_modelicac)
 	$(OCAMLC) -o $@ -I $(BASE_PATH_MODELICAC) $(OCAMLOPTFLAGS) $(OCAMLLIBS) $(MLS_modelicac)
 
 XML2Modelica: $(XML2Modelica_SOURCES) $(MLS_XML2Modelica)
-	$(OCAMLOPT) -o $@ -I $(BASE_PATH_XML2Modelica) $(OCAMLCFLAGS) $(OCAMLLIBS:.cma=.cmxa) $(MLS_XML2Modelica)
+	ocamlfind ocamlopt -package num -linkpkg -o $@ -I $(BASE_PATH_XML2Modelica) $(OCAMLCFLAGS) $(MLS_XML2Modelica)
 
 XML2Modelica.ocamlc: $(XML2Modelica_SOURCES) $(MLS_XML2Modelica)
 	$(OCAMLC) -o $@ -I $(BASE_PATH_XML2Modelica) $(OCAMLCFLAGS) $(OCAMLLIBS) $(MLS_XML2Modelica)
 
 modelicat: $(modelicat_SOURCES) $(MLS_modelicat)
-	$(OCAMLOPT) -o $@ $(INCLUDE_modelicat) $(OCAMLOPTFLAGS) $(OCAMLLIBS:.cma=.cmxa) $(MLS_modelicat)
+	ocamlfind ocamlopt -package num -linkpkg -o $@ $(INCLUDE_modelicat) $(OCAMLOPTFLAGS) $(MLS_modelicat)
 
 modelicat.ocamlc: $(modelicat_SOURCES) $(MLS_modelicat)
 	$(OCAMLC) -o $@ $(INCLUDE_modelicat) $(OCAMLOPTFLAGS) $(OCAMLLIBS) $(MLS_modelicat)
